cf-check/dump.c

```diff
@@ -299,15 +299,15 @@ static void print_struct_or_string(
                 print_struct_averages(value, strip_strings, tskey_filename);
             }
         }
-        else if (StringEndsWith(file, "history.lmdb"))
+        else if (StringContains(file, "history.lmdb"))
```

Comment:
I'm guessing `file` could be a path (with slashes) as well. So in addition to the example you gave in the commit. It could be `..../our_special_history.lmdb.d/..../cf_lock.lmdb`.

What was there already is not great. I could easily see the introduction of a `reporting_history.lmdb` or something, breaking both the old and new code.

IMO, the ideal approach would be something quite careful and specific:

```
StringEqual(basename(file), "history.lmdb") || StringEqual(basename(file), "history.lmdb.backup")
```

But these are quite rare edge cases, so it's not a big deal.

Reply:
Yes, for this particular LMDB file it makes sense, I guess. The others are more specific. Replacing this with
```
        else if (StringEqual("history.lmdb") || StringEndsWith(file, FILE_SEPARATOR"history.lmdb") ||
                 StringEqual("history.lmdb.backup") ||
                 StringEndsWith(file, FILE_SEPARATOR"history.lmdb.backup"))
```
to avoid `basename()` which is a PITA.

Reply:
But a proper answer here is that we should have code that identifies our DBs when given a file path, potentially even checking the contents/metadata/whatever.